src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -265,6 +229,20 @@ void GinIndexStore::finalize()
     if (!current_postings.empty())
         writeSegment();
 
+    // If next available segment id has value, then the value has to be saved to disk again with the updated segment id
```

Comment:
Are we really saving the value "again"? Maybe just remove the comment to avoid confusion, I think it is clear what happens.